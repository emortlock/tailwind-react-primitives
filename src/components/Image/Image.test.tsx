import React from 'react'
import { render, RenderResult } from '@testing-library/react'

import { Image } from '.'

const setup = (
  testProps = {},
): [RenderResult, { props: Record<string, any> }] => {
  const props = Object.assign({ src: '/static/assets/tubgirl.gif' }, testProps)

  const wrapper = render(<Image {...props} />)

  return [
    wrapper,
    {
      props,
    },
  ]
}

describe('Image', () => {
  it('renders matching snapshot', () => {
    const [{ asFragment }] = setup()

    expect(asFragment()).toMatchSnapshot()
  })
})
