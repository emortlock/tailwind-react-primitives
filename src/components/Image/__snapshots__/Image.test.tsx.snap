// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders matching snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative w-full"
  >
    <div
      class=""
      style="padding-bottom: 100%;"
    />
    <img
      class="absolute inset-0 w-full"
      src="/static/assets/tubgirl.gif"
    />
  </div>
</DocumentFragment>
`;
